File: euler_10.py
#!/usr/bin/env python3

'''
Summation of primes
Problem 10

The sum of the primes below 10 is 2 + 3 + 5 + 7 = 17.

Find the sum of all the primes below two million.
'''

from primesieve import primes

print(sum(primes(2000000)))

'''
from math import sqrt

def check_primeness(x):
    for i in range(2, int(sqrt(x))+1):
        if (x % i == 0):
            return False
    return True

my_sum = 0

for i in range(2, 2000000):
    if check_primeness(i):
        my_sum += i

print(my_sum)
'''

File: euler_10.js
#!/usr/bin/js

/*
Summation of primes
Problem 10

The sum of the primes below 10 is 2 + 3 + 5 + 7 = 17.

Find the sum of all the primes below two million.
*/

function checkPrimeness(x) {
    for (var i = 2; i <= Math.floor(Math.sqrt(x)); i++) {
        if (x % i === 0) {
            return false;
        }
    }
    return true;
}

var mySum = 2;

for (var i = 3; i < 2000000; i = i+2) {
    if (checkPrimeness(i)) {
        mySum += i;
    }
}

console.log(mySum);
